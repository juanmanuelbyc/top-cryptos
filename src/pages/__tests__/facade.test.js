import React from 'react';
import renderer from 'react-test-renderer';
import Facade from '../../components/facade';
import { Router } from 'react-router-dom';
import {createMemoryHistory} from 'history';

const TestFacade = () => {
 const history = createMemoryHistory();
 return (
     <Router location={history.location} navigator={history}>
       <Facade image={'../assets/cryptoImage.png'} topText={'top text'} mainText={'main text'} bottomText={'bottom text'} />
     </Router>
 );
};

it('renders correctly', () => {
  const tree = renderer.create(
    <TestFacade />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
