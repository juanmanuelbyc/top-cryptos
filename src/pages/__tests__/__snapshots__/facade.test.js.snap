// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="facade"
>
  <div
    className="facadeTopSection"
  >
    <a
      className="backHomeButton"
      href="/Home"
      onClick={[Function]}
    >
      ‚Üê
    </a>
    <p
      className="facadeTopText"
    >
      top text
    </p>
    <div
      className="topFacadeIcons"
    >
      <img
        alt="mic Icon"
        className="micIcon"
        src="micIcon.png"
      />
      <img
        alt="settings Icon"
        className="settingsIcon"
        src="settingsIcon.png"
      />
    </div>
  </div>
  <div
    className="facadeMiddleSection"
  >
    <div
      className="facadeImageContainer"
    >
      <img
        alt="cryptoImage"
        className="facadeImage"
        src="../assets/cryptoImage.png"
      />
    </div>
    <div
      className="facadeTextsContainer"
    >
      <h2
        className="facadeMainText"
      >
        main text
      </h2>
    </div>
  </div>
  <div
    className="facadeBottomSection"
  >
    <p>
      bottom text
    </p>
  </div>
</div>
`;
